core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1927,4 +1927,31 @@ package object config {
       .version("3.0.1")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val PUSH_BASED_SHUFFLE_ENABLED =
+    ConfigBuilder("spark.shuffle.push.based.enabled")
+      .doc("Set to 'true' to enable push based shuffle")
+      .booleanConf
+      .createWithDefault(false)
+
+  private[spark] val MAX_MERGER_LOCATIONS_CACHED =
```

Comment:
This configuration is not used?
Seems that you didn't add `maxRetainedMergerLocations` in `BlockManagerMasterEndpoint`.
That should be included with this PR.